libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/notification/impl/EmailNotificationService.java

```diff
@@ -117,14 +118,30 @@
         athenzUIUrl = System.getProperty(PROP_NOTIFICATION_ATHENZ_UI_URL);
         from = System.getProperty(PROP_NOTIFICATION_EMAIL_FROM);
 
-        logoImageResource = getClass().getClassLoader().getResource(EMAIL_TEMPLATE_ATHENZ_LOGO);
+        logoImage = readBinaryFromFile(EMAIL_TEMPLATE_ATHENZ_LOGO);
         emailBaseCSS = readContentFromFile(EMAIL_TEMPLATE_CSS);
         emailMembershipApprovalBody = readContentFromFile(EMAIL_TEMPLATE_NOTIFICATION_APPROVAL);
         emailMembershipApprovalReminderBody = readContentFromFile(EMAIL_TEMPLATE_NOTIFICATION_APPROVAL_REMINDER);
         emailDomainMemberExpiryBody = readContentFromFile(EMAIL_TEMPLATE_DOMAIN_MEMBER_EXPIRY);
         emailPrincipalExpiryBody =  readContentFromFile(EMAIL_TEMPLATE_PRINCIPAL_EXPIRY);
     }
 
+    byte[] readBinaryFromFile(String fileName) {
+
+        byte[] fileByteArray = null;
+        URL resource = getClass().getClassLoader().getResource(fileName);
+        if (resource != null) {
+            try (InputStream fileStream = getClass().getClassLoader().getResourceAsStream(fileName)) {
+                //convert to byte array
+                fileByteArray = IOUtils.toByteArray(Objects.requireNonNull(fileStream));
```

Comment:
Both here and in the readContentFromFile method why are we calling Objects.requireNonNull if we have already verified that the resource is valid - there is resource != null check? was the plan to remove the initial resource check completely?